todomvrx/src/main/java/com/airbnb/mvrx/todomvrx/views/AddEditView.kt

```diff
@@ -1,36 +1,115 @@
 package com.airbnb.mvrx.todomvrx.views
 
 import android.content.Context
+import android.text.Editable
+import android.text.Spanned
+import android.text.TextWatcher
 import android.util.AttributeSet
 import android.widget.EditText
 import android.widget.ScrollView
+import com.airbnb.epoxy.CallbackProp
 import com.airbnb.epoxy.ModelView
 import com.airbnb.epoxy.TextProp
 import com.airbnb.mvrx.todomvrx.todoapp.R
 
+
 @ModelView(autoLayout = ModelView.Size.MATCH_WIDTH_MATCH_HEIGHT)
 class AddEditView @JvmOverloads constructor(
-    context: Context,
-    attrs: AttributeSet? = null,
-    defStyleAttr: Int = 0
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0
 ) : ScrollView(context, attrs, defStyleAttr) {
 
     private val titleView by lazy { findViewById<EditText>(R.id.task_title) }
     private val descriptionView by lazy { findViewById<EditText>(R.id.task_description) }
 
+    private val titleWatcher = SimpleTextWatcher { onTitleChanged?.invoke(it) }
+    private val descriptionWatcher = SimpleTextWatcher { onDescriptionChanged?.invoke(it) }
+
     init {
         inflate(context, R.layout.add_edit_view, this)
+        titleView.addTextChangedListener(titleWatcher)
+        descriptionView.addTextChangedListener(descriptionWatcher)
     }
 
     @TextProp
     fun setTitle(title: CharSequence?) {
-        titleView.setText(title)
+        titleView.setTextIfDifferent(title)
     }
 
     @TextProp
     fun setDescription(description: CharSequence?) {
-        descriptionView.setText(description)
+        descriptionView.setTextIfDifferent(description)
+    }
+
+    @set:CallbackProp
+    var onTitleChanged: ((newText: String) -> Unit)? = null
+
+    @set:CallbackProp
+    var onDescriptionChanged: ((newText: String) -> Unit)? = null
+
+}
+
+/**
+ * Set the given text on the textview.
+ *
+ * @return True if the given text is different from the previous text on the textview.
+ */
+fun EditText.setTextIfDifferent(newText: CharSequence?): Boolean {
```

Comment:
I've actually considered making a library of idempotent base classes like TextViews, ImageViews, etc with this built in.

Reply:
That would be great, people ask about that for Epoxy a lot